lib/em/pure_ruby.rb

```diff
@@ -927,32 +1005,71 @@ def self.connect bind_addr, bind_port, host, port
     def initialize io
       super
       @pending = true
-      @handshake_complete = false
+      @ssl_handshake_state = nil
     end
 
+    TCP_ESTABLISHED = 1 # why isn't this already a const in Socket?
+
     def ready?
-      if RUBY_PLATFORM =~ /linux/
-        io.getsockopt(Socket::SOL_TCP, Socket::TCP_INFO).unpack("i").first == 1 # TCP_ESTABLISHED
+      if defined?(Socket::SOL_TCP) && defined?(Socket::TCP_INFO)
```

Comment:
Nice

Reply:
Yeah, unfortunately, my efforts didn't result in a fully working version for windows and macos.  I'd say that there's still some refactoring to do in that area of the pure ruby reactor to bring it to parity with the cpp version on all platforms.